src/Projection/ProjectionValidator.php

```diff
@@ -0,0 +1,18 @@
+<?php declare(strict_types=1);
+
+namespace Comquer\Projection;
+
+use Comquer\Validation\ArrayValidator;
+use Comquer\Validation\ArrayMissingRequiredKeysException;
+
+class ProjectionValidator
+{
+    public static function validateSerialized(string $projectionName, array $requiredKeys, array $serializedProjection): void
```

Comment:
I dont fancy this approach, static methods are good as a static factory methods, if you start calling them like functions or just putting them in classes for organisation it kinda looks bad. Consider refactoring this to real class which implements some kind of validator interface and can execute validate method for better OOP experience.

Reply:
Perhaps an interface would be an overkill here but I agree that making the methods dynamic is a safer and overall better solution.